kedro/io/data_catalog.py

```diff
@@ -94,6 +96,36 @@ def _sub_nonword_chars(data_set_name: str) -> str:
     return re.sub(WORDS_REGEX_PATTERN, "__", data_set_name)
 
 
+def _specificity(pattern: str) -> int:
+    """Helper function to check the length of exactly matched characters not inside brackets
+    Example -
+    specificity("{namespace}.companies") = 10
+    specificity("{namespace}.{dataset}") = 1
+    specificity("france.companies") = 16
+    Args:
+        pattern: The factory pattern
+    """
+    # Remove all the placeholders from the pattern
+    result = re.sub(r"\{.*?\}", "", pattern)
+    return len(result)
+
+
+def _sort_dataset_factory_patterns(dataset_patterns: list) -> list:
```

Comment:
Having helper functions outside of the class makes inheritance a bit harder. Can you add this one and the previous one as a class method to the `DataCatalog`?